test/NuGet.Core.FuncTests/Dotnet.Integration.Test/ProjectPropertyTests.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+using FluentAssertions;
+using NuGet.Test.Utility;
+using Xunit;
+
+namespace Dotnet.Integration.Test
+{
+    [Collection(DotnetIntegrationCollection.Name)]
+    public class ProjectPropertyTests
+    {
+        private MsbuildIntegrationTestFixture _msbuildFixture;
+
+        public ProjectPropertyTests(MsbuildIntegrationTestFixture fixture)
+        {
+            _msbuildFixture = fixture;
+        }
+
+        [Fact]
+        public void NuGetAudit_EnabledByDefaultOnNet8AndHigher()
```

Comment:
I think this should be a unit test, these integration/functional tests are slow and are good for testing something end-to-end like restore, build, and pack as one operation.  What do you think about just making it a unit test?

Reply:
Sure, I'd prefer a unit test, but I'm not really sure how to design it. When I started this PR, I was thinking I really want to have the patched SDK evaluate a project file.  But I guess if we load the NuGet.targets file via MSBuild APIs, and set NETCoreSdkVersion as a global variable, in order to "mock" the SDK, then that's good enough?

Otherwise my only other idea is to use an MSBuild Task with an output, but that's really overkill for what without unit testing is achieved with a single property with a condition.

However, looking at the CI failures, particularly the mono tests failing because the MSBuild intrinsic function to compare versions doesn't exist, is going to make adding this rather difficult.

Reply:
Yeah it looks like you'll want to use `$([MSBuild]::ValueOrDefault())`:

```
$([MSBuild]::VersionGreaterThanOrEquals($([MSBuild]::ValueOrDefault('$(NETCoreSdkVersion)', '1.0')), '8.0'))
```

Your unit tests could check what happens if `NETCoreSdkVersion` isn't set, its set to lower than 8.0, and higher than 8.0, as well as if the value for `NuGetAudit` is left alone if set by the user.  But I don't see any unit tests that use MSBuild APIs to evaluate a project?

Reply:
I had another go at trying to write a test using MSBuild APIs to load the targets file (in NuGet.Build.Tasks.Test). Trying to load the project fails at runtime due to Microsoft.Build.dll not being found. So, I tried to use Microsoft.Build.Locator, but now xUnit won't load the test assembly because the dependant (MSBuild) assemblies can't be found.

So, for now I'm giving up on trying to use MSBuild APIs to load the targets file directly. The dotnet integration test is slow, but it works.